borg/upgrader.py

```diff
@@ -267,10 +267,14 @@ def find_key_file(cls, repository):
         get_keys_dir = cls.get_keys_dir
         id = hexlify(repository.id).decode('ascii')
         keys_dir = get_keys_dir()
+        def ex():
+            raise KeyfileNotFoundError(repository.path, keys_dir)
```

Comment:
I hate having to duplicate code for the same control flow, but if you wish, so be it.

I’d add test but I have no idea how to get to those branches without heavy patching.